### ISSUE TITLE
As a user I want to change the file paths of products in the registry

### ISSUE THREAD

**Issue by mdrum** [NASA-PDS/registry#96]
## 💪 Motivation
...so that I can keep the registry up to date if files move/migrate on my archive server.

## 📖 Additional Details
These two fields in the registry can become outdated if files change their locations for one reason or another:

1. ops:Data_File_Info/ops:file_ref
2. ops:Label_File_Info/ops:file_ref

If files move within my archive and I run the harvest tool on that bundle again, the changes to the file paths are not pushed up to the registry because they are skipped (it looks up each LIDVID, sees that it's already registered, and skips any other processing).

It's possible to update the individual registry docs via the OpenSearch API, but it's actually quite complicated to do so for more than one doc in one query. This is because bulk updates must be provided with a new value for every matching doc, but in this case the values would all need to be different for each product (they all have different file _names_). You can provide a script as part of the query that can be run, but I wrestled with the weird half-java and regex syntax for a few hours and never got through all the error messages. This leaves the only workaround to instead write a script that will generate a series of API queries for each individual product, but that sounds like what the Registry Manager is already supposed to be good at.

## ⚖️ Acceptance Criteria
Either:
1. There exists a command that can perform a prefix-replace on the two file-path fields in the registry. For example, I could run something like `registry-manager update-data-location -prefix https://sbnarchive.psi.edu/pds4/orex/ -replaceWith https://sbnarchive.psi.edu/pds4/orex_v10/` and it would query my entire registry and replace any Label or Data file refs with the amended prefix.
2. Alternatively, if Harvest/Crawler could instead perform some additional checks when it running a harvest job to not just check if a LIDVID is registered, but that its file path (and checksum?) has not changed. If it had, it would perform an update of those fields.

**Comment by al-niessner** [comment]
@jordanpadams 

I am confused by this one. How is this different than harvesting with force or overwrite (whichever it is)? I get that you do not want to redo a whole bundle, so harvest just the one file.

Rather than make registry-mgr all confused with this, I would say that harvest take a list of label paths to process. The force/overwrite would then do the trick.

**Comment by jordanpadams** [comment]
@al-niessner I agree with you that you can just re-harvest the data, but users have asked for a slightly alternate solution to just updating paths to hopefully improve performance. It may be we just operate harvest exactly the same, but we only update the file path for the products, versus overwriting the whole doc.

**Comment by jordanpadams** [comment]
I could see this being a harvest function as well if that makes the most sense.

**Comment by al-niessner** [comment]
@jordanpadams 

Are you saying that they would be opposed to a file that listed the labels they wanted to update rather than running the bundle? I mean they have to update the label to point at the new location anyway. I maybe the label moved with it? But then the collection would need updated. No matter how I try to work out a scenario where files get shuffled but he keeper of the label is unaware just does not work. The only reason to not want to make file of changed/moved labels is effort?

The other problem I am having, if you moved a file and changed the label contents you probably did more than just the path. Maybe I am wrong and there is a whole workforce doing nothing but shuffling data files to stay employed, but it really seems like there is something more happening. Data compression for example.

**Comment by jordanpadams** [comment]
@al-niessner I think we will need to go to the users with a d fewesign proposals to figure out the best path forward. Here are a couple ideas. We can gather 1-2 more and bring them to the SWG for input.

1. a table of something like `lidvid, new ops:Data_File_Info.ops:file_ref, new ops:Label_File_Info.ops:file_ref`
2. point harvest at some directory with some flag that says "change path only" and harvest will read all the labels and only update the associated file paths

In PDS world, folks often move files around when they are superseded, requiring a path change.

That being said, part of the parent Epic associated with this ticket is handling what you are talking about. Not only is the path to the file_ref different, but they actually rename the file separate from what the label says. That is where I feel like solution (1) above would work for both of these use cases.

**Comment by al-niessner** [comment]
@jordanpadams 

Idea 3: give harvest a list of labels (file paths) to load and they can add not add the overwrite as necessary.

idea 4: give harvest a bundle but only do these lidvids. Slooowww.

**Comment by al-niessner** [comment]
@jordanpadams 

Cannot do 1 because if there are more than 1 file_ref, there is no way to match it unless you meant `lidvid, old ref, new ref`.

**Comment by jordanpadams** [comment]
@al-niessner right. I meant `lidvid, new label ref, new data ref(s)`. More generally, I am thinking we give users the ability to overwrite the value(s) for a specific field, or if the value does not exist, add it to the document.

**Comment by al-niessner** [comment]
@jordanpadams 

Decided to look at harvest configuration. Can already do this
```
<harvest>
  <load>
    <fiiles>
      <manifest>/moved/file/a</manifest>
      <manifest>/moved/file/b</manifest>
      <manifest>/moved/file/c</manifest>
    </files>
  </load>
  blah.blah.blah.from.bundle.load
</harvest>
```

Then run `harvest --overwrite`. Does just want you want. Yeah, it does more than just ref, but it gets ref done and anything else that needs it. Since only doing 10 items, should be fast. Way faster than anything else.

**Comment by al-niessner** [comment]
@jordanpadams 

Um, why are adding random fields to a document? I thought that the document in the database was supposed to represent the one outside the database. Adding random stuff kind of breaks that in a big way. Still going back to using harvest. Apparently you are thinking of something else entirely and just are not voicing it.

If you want, I can add a `registry-mgr butcher-db -key some_random_text -value some_random_value` that will allow the user to add or modify (if they type it right) key with any value. I know I am being snarky, but I am trying to make a point on what you are suggesting.
